modules/security-lti/src/main/java/org/opencastproject/security/lti/LtiLaunchAuthenticationHandler.java

```diff
@@ -137,6 +137,9 @@
   /** Set of usernames that should not authenticated as themselves even if the OAuth consumer keys is trusted */
   private Set<String> usernameBlacklist = new HashSet<>();
 
+  /** concurrent attemtps */
+  private Map<String, Boolean> attempts = new ConcurrentHashMap<>(128);
```

Comment:
Purely nitpicking but it's not that we count attempts so the choice for the name seems a bit odd.
;aybe:

```suggestion
  private Map<String, Boolean> activePersistence = new ConcurrentHashMap<>(128);
```